src/router.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash 
+
+# Common pipeline build script will 
+# delegate to appropriate composer/fabric build script
+# script receives two parameters, the stage to route  
+# and the platform 
+
+stage="$1"
+platform="$2"
+executable_script=""
+
+if [[ "${stage}" != "build" ]] && [[ "${stage}" != "test" ]] && [[ "${stage}" != "deploy" ]]; then
+    echo "Invalid stage: ${stage} selected"
+    exit 1
+fi
+
+echo "${stage} stage selected"
+
+if [ "${platform}" = "go" ]; then
```

Comment:
I would prefer to see a router script common rather than being at the top level like this. For an example, have a look how https://github.com/hyperledger/composer-tools/blob/master/packages/fabric-dev-servers/stopFabric.sh routes to the correct script using https://github.com/hyperledger/composer-tools/blob/master/packages/fabric-dev-servers/_loader.sh
That keeps the straight forward, and hopefully more stable, build/test/deploy script 'API' for pipelines to use, while keeping too much duplication out of those top level scripts

Reply:
I think that  router and any actual executable script called directly by a toolchain stage should be placed  in this top level directory.  Just like prepare.sh.   Common in my opinion should be reserved more for library type  of scripts that can be sourced in other scripts.

Reply:
I'd prefer to keep a clear distinction between build/pipeline scripts that appear in templates, or could be used in other ci environments, and 'plumbing' scripts.